app/jobs/geocode_location_job.rb

```diff
@@ -47,6 +51,6 @@ def enqueue_next_job
   end
 
   def next_location
-    Location.where('latitude IS NULL OR longitude IS NULL').first
+    Location.where(geocode_response: nil).first
```

Comment:
Yeah, I think a constant might be a good solution here. Rather than having an exception class, maybe a string value to reflect the conditions? Would it be worth putting a `not null` on that column?